assets/extensions/store.js

```diff
@@ -0,0 +1,292 @@
+				method: 'POST',
+				data: { plugins: slugs },
+			} );
+
+			yield actions.setError( null );
+			yield actions.setExtensions(
+				{
+					extensions: response.completed,
+					wccom_connected: response.wccom_connected,
+				},
+				true
+			);
+
+			yield dispatch( 'core/notices' ).createNotice(
+				'success',
+				__( 'Update completed succesfully!', 'sensei-lms' ),
+				{
+					type: 'snackbar',
+				}
+			);
+		} catch ( error ) {
+			yield actions.setExtensionsStatus( slugs, '' );
+			yield actions.setError(
+				sprintf(
+					// translators: Placeholder is underlying error message.
+					__(
+						'There was an error while updating the plugin: %1$s.',
+						'sensei-lms'
+					),
+					error.message
+				)
+			);
+		} finally {
+			// Update queue extensions, if exists.
+			const queuedExtensions = yield select(
+				EXTENSIONS_STORE
+			).getExtensionsByStatus( STATUS.IN_QUEUE );
+
+			if ( 0 === queuedExtensions.length ) {
+				return;
+			}
+
+			yield actions.updateExtensions( queuedExtensions );
+		}
+	},
+
+	/**
+	 * Set extensions in progress.
+	 *
+	 * @param {string} slugs  Extensions in progress.
+	 * @param {string} status Status.
+	 */
+	setExtensionsStatus( slugs, status ) {
+		return {
+			type: 'SET_EXTENSIONS_STATUS',
+			slugs,
+			status,
+		};
+	},
+
+	/**
+	 * Set the extensions layout.
+	 *
+	 * @param {Object} obj        Layout object.
+	 * @param {Array}  obj.layout Extensions layout.
+	 */
+	setLayout( { layout = [] } ) {
+		return {
+			type: 'SET_LAYOUT',
+			layout,
+		};
+	},
+
+	/**
+	 * Set the error message.
+	 *
+	 * @param {string} error The error.
+	 */
+	setError( error ) {
+		return {
+			type: 'SET_ERROR',
+			error,
+		};
+	},
+};
+
+/**
+ * Extension store selectors.
+ */
+const selectors = {
+	getExtensions: ( { extensions, entities } ) =>
+		extensions.map( ( slug ) => entities.extensions[ slug ] ),
+	getExtensionsByStatus: ( args, status ) =>
+		selectors
+			.getExtensions( args )
+			.filter( ( extension ) => status === extension.status ),
+	getLayout: ( { layout } ) => layout,
+	getError: ( { error } ) => error,
+};
+
+/**
+ * Extension store resolvers.
+ */
+const resolvers = {
+	/**
+	 * Loads the extensions during initialization.
+	 */
+	*getExtensions() {
+		const response = yield apiFetch( {
+			path: '/sensei-internal/v1/sensei-extensions?type=plugin',
+		} );
+
+		return actions.setExtensions( response );
+	},
+
+	/**
+	 * Loads the extensions layout.
+	 */
+	*getLayout() {
+		const response = yield apiFetch( {
+			path: '/sensei-internal/v1/sensei-extensions/layout',
+		} );
+
+		return actions.setLayout( response );
+	},
+};
+
+/**
+ * Extensions store reducer.
+ *
+ * @param {Object} state  The store state.
+ * @param {Object} action The action to handle.
+ */
+const reducer = (
+	state = {
+		extensions: [],
+		entities: { extensions: {} },
+		layout: [],
+		error: null,
+	},
+	action
+) => {
+	switch ( action.type ) {
+		case 'SET_EXTENSIONS':
```

Comment:
Could wrap this with `createReducerFromActionMap` like the other stores to get rid of the switch stuff.

Reply:
Refactor here: https://github.com/Automattic/sensei/pull/4205/commits/57352f7a5067e2604a67fe4525ad039930390937